easybuild/easyblocks/o/openssl_wrapper.py

```diff
@@ -0,0 +1,268 @@
+            opensslv_path = os.path.join(include_dir, 'opensslv.h')
+            if os.path.exists(opensslv_path):
+                opensslv = read_file(opensslv_path)
+                header_majmin_version = re.search(r"SHLIB_VERSION_NUMBER\s\"([0-9]+\.[0-9]+)", opensslv, re.M)
+                if re.match("^{}".format(*header_majmin_version.groups()), self.version):
+                    self.ssl_sysheader = include_dir
+                    self.log.info("Found OpenSSL headers in host system: %s", self.ssl_sysheader)
+                    break
+
+        if not self.ssl_sysheader:
+            self.log.info("OpenSSL headers not found in host system")
+
+    def fetch_step(self, *args, **kwargs):
+        """Fetch sources if OpenSSL component is needed"""
+        if not all([self.ssl_syslib, self.ssl_sysheader]):
+            super(EB_OpenSSL_wrapper, self).fetch_step(*args, **kwargs)
+
+    def extract_step(self):
+        """Extract sources if OpenSSL component is needed"""
+        if not all([self.ssl_syslib, self.ssl_sysheader]):
+            super(EB_OpenSSL_wrapper, self).extract_step()
+
+    def install_step(self):
+        """Symlink target OpenSSL installation"""
+        shlib_ext = get_shared_lib_ext()
+
+        if self.ssl_syslib and self.ssl_sysheader:
+            # The host system already provides the necessary OpenSSL files
+            lib_pattern = ['%s*.%s*' % (lib_so, shlib_ext) for lib_so in self.openssl_libs]
+            lib_pattern = [os.path.join(os.path.dirname(self.ssl_syslib), '%s' % ptrn) for ptrn in lib_pattern]
+            lib_pattern.append(os.path.join(self.ssl_sysengines, '*'))
+
+            include_pattern = [os.path.join(self.ssl_sysheader, '*')]
+
+            bin_path = which(self.name.lower())
+
+            # Link OpenSSL libraries
+            lib64_dir = os.path.join(self.installdir, 'lib64')
+            lib64_engines_dir = os.path.join(lib64_dir, os.path.basename(self.ssl_sysengines))
+            mkdir(lib64_engines_dir, parents=True)
+            symlink('lib64', 'lib')
+
+            lib_files = expand_glob_paths(lib_pattern)
+
+            for libso in lib_files:
+                if 'engines' in libso:
+                    target_dir = lib64_engines_dir
+                else:
+                    target_dir = lib64_dir
+                symlink(libso, os.path.join(target_dir, os.path.basename(libso)))
+
+            # Link OpenSSL headers
+            include_dir = os.path.join(self.installdir, 'include', self.name.lower())
+            mkdir(include_dir, parents=True)
+
+            include_files = expand_glob_paths(include_pattern)
+            for header in include_files:
+                symlink(header, os.path.join(include_dir, os.path.basename(header)))
+
+            # Link OpenSSL binary
+            bin_dir = os.path.join(self.installdir, 'bin')
+            mkdir(bin_dir)
+            symlink(bin_path, os.path.join(bin_dir, os.path.basename(bin_path)))
```

Comment:
```suggestion
            symlink(bin_path, os.path.join(bin_dir, os.self.name.lower()))
```

Reply:
`os.self.name` -> `self.name` :)

Reply:
@Micket `openssl11` binaries fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2429/commits/25d718c26bc846c5893cb505c0c491820423570d